jaxtyping/array_types.py

```diff
@@ -384,7 +384,9 @@ def __getitem__(cls, item: Tuple[Any, str]) -> _MetaAbstractArray:
                 elem = _SymbolicDim(elem, broadcastable)
             dims.append(elem)
         if _array_name_format == "dtype_and_shape":
-            name = f"{cls.__name__}[{array_type.__name__}, '{dim_str}']"
+            # In python 3.8, e.g., typing.Union lacks `__name__`.
+            type_str = getattr(array_type, '__name__', None) or repr(array_type)
```

Comment:
I'm not a fan of the implicit-cast-to-bool. Can this be:
```
try:
    type_str = array_type.__name__
except AttributeError:
    type_str = repr(array_type)
```

Reply:
Done.